docs/js-demo/edit.js

```diff
@@ -195,44 +196,50 @@ myCode1Mirror.on("changes", onEditChanges);
 
 jsCode1Mirror.setSize(null,codeMirrorDefaultHeight);
 
-
-
 //checks or unchecks the eval button
 function changeEvalButton(bool) {
   $('#option-eval').prop('checked', bool);
   onEvalButtonChange();
 }
 
-//creates a gist from OCaml code
+//creates a hashed URL from OCaml code IF encoded hash is under 1,950 characters
+//otherwise creates a github gist
 $('#share').click(function (e) {
-  var state = $(this).button('loading');
-  var request =
-  {
-    "description": "BuckleScript Gist",
-    "public": true,
-    "files": {
-      "gist.ml": {
-        "content": myCode1Mirror.getValue()
+  var hashedCode = btoa(myCode1Mirror.getValue());
+  if (hashedCode.length <= 1950) {
+    var url = 'https://bloomberg.github.io/bucklescript/js-demo/#' + hashedCode;
```

Comment:
can we encode it explicitly whether it is base64 encoded or not instead of relying on a length check?
I think you can already tell the difference without relying on a magic number by checking `?gist=`